packages/headless/src/features/commerce/context/cart/cart-actions.ts

```diff
@@ -0,0 +1,58 @@
+import { createAction } from '@reduxjs/toolkit';
+import {validatePayload} from '../../../../utils/validate-payload';
+import {CartItemParam} from '../../../../api/commerce/commerce-api-params';
+import { ArrayValue, NumberValue, RecordValue } from '@coveo/bueno';
+
+export interface SetCartPayload {
+  cart: CartItemParam[]
+}
+export const setCart = createAction(
+  'commerce/cart/set',
+  (payload: SetCartPayload) =>
+    validatePayload(payload, {
+      cart: new ArrayValue({
```

Comment:
This schema seems a copy of this of what's already in headless-cart.ts.

Since it's a complex one, it would be worth it to define it here as a public module variable and reuse in both files.